sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -348,7 +312,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
    * that are not present in the productIterator.
    * @param newArgs the new product arguments.
    */
-  def makeCopy(newArgs: Array[AnyRef]): this.type = attachTree(this, "makeCopy") {
+  def makeCopy(newArgs: Array[AnyRef]): BaseType = attachTree(this, "makeCopy") {
```

Comment:
Sorry, this is mostly unrelated.  However, using `this.type` is invalid here, as it does not mean "the type of this object" but instead "the singleton that is this specific object".  Since the JVM has no such concept, the code before worked... but it seemed better to remove it if I could.  In other places I left it since they would require code changes elsewhere in SQL.